obspy/clients/httpproxy.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+"""
+Establish a socket connection through an HTTP proxy.
+Author: Fredrik Østrem <frx.apps@gmail.com>
+License:
+  This code can be used, modified and distributed freely, as long as it is this
+  note containing the original author, the source and this license, is put
+  along with the source code.
+
+J. MacCarthy, modified from https://gist.github.com/frxstrem/4487802
+
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+
+import socket
+from base64 import b64encode
+import urllib
+import urlparse
+
+
+def get_proxy_tuple():
+    """
+    Return system http proxy as a urlparse.urlparse tuple or () if unset.
+    """
+    proxydict = urllib.getproxies()
+    proxystr = proxydict.get('http') or proxydict.get('https') or ''
+    if proxystr:
+        proxy = urlparse.urlparse(proxystr)
+    else:
+        proxy = ()
+
+    return proxy
+
+
+def valid_address(addr):
+    """ Verify that an IP/port tuple is valid """
+    is_valid = (isinstance(addr, (list, tuple)) and
+                len(addr) == 2 and
+                isinstance(addr[0].decode(), (str, unicode)) and
```

Comment:
The `.decode()` does no longer exist in the latest version. Github now also shows outdated code snippets...